packages/@azure/servicebus/data-plane/lib/queueClient.ts

```diff
@@ -166,16 +166,18 @@ export class QueueClient extends Client {
    */
   async getSessionReceiver(options?: SessionReceiverOptions): Promise<SessionReceiver> {
     if (!options) options = {};
-    if (
-      options.sessionId &&
-      this._context.messageSessions[options.sessionId] &&
-      this._context.messageSessions[options.sessionId].isOpen()
-    ) {
-      throw new Error(
-        `Close the current session receiver for sessionId ${
-          options.sessionId
-        } before using "getSessionReceiver" to create a new one for the same sessionId`
-      );
+    if (options.sessionId) {
+      if (
+        this._context.messageSessions[options.sessionId] &&
+        this._context.messageSessions[options.sessionId].isOpen()
+      ) {
+        throw new Error(
+          `Close the current session receiver for sessionId ${
+            options.sessionId
+          } before using "getSessionReceiver" to create a new one for the same sessionId`
+        );
+      }
+      delete this._context.expiredMessageSessions[options.sessionId];
```

Comment:
Another place where we can clear this cache is when the user explicitly closes the session receiver.
Thoughts?

Reply:
And when we close the queueClient/subscriptionClient

Reply:
It's for the same reason that we had initially discussed - we'll need to have it in the cache in case user subsequently attempts to access the session.

Reply:
There are mainly 2 ways the `close()` on the `messageSession` object is called
- One is when the user intentionally tries to close the namespace/queueClient/subscriptionClient/sessionReceiver
- The other is when we (the sdk) closes it due to an internal event i.e the like receiver error or timeouts

I wonder if we should treat the user subsequently attempting to engage with a message in the above 2 cases separately

Reply:
As discussed offline,
- Lets clear the `this._context.expiredMessageSessions` inside `_onAmqpClose()` as long as the error is not SessionLockLostError
- Lets look into using `ConcurrentExpiringMap` for maintaining this cache of `expiredMessageSessions`
- Lets improve the language in `Cannot find the receiver with name...`